test/steps/v2_steps.py

```diff
@@ -2457,23 +2603,74 @@ def add_abi_method_call(context, account_type, operation):
         raise NotImplementedError(
             "cannot make transaction signer for " + account_type
         )
-    app_args = process_abi_args(context.abi_method, context.method_args)
+    approval_program = clear_program = None
+    global_schema = local_schema = None
+
+    def int_if_given(given):
+        return int(given) if given else 0
+
+    local_schema = global_schema = None
+    if create_when_calling:
+        if approval_program_path:
+            approval_program = read_program(context, approval_program_path)
+        if clear_program_path:
+            clear_program = read_program(context, clear_program_path)
+        if local_ints or local_bytes:
+            local_schema = transaction.StateSchema(
+                num_uints=int_if_given(local_ints),
+                num_byte_slices=int_if_given(local_bytes),
+            )
+        if global_ints or global_bytes:
+            global_schema = transaction.StateSchema(
+                num_uints=int_if_given(global_ints),
+                num_byte_slices=int_if_given(global_bytes),
+            )
+        extra_pages = int_if_given(extra_pages)
+
+    app_id = int(context.current_application_id)
+
+    app_args = process_abi_args(
+        context, context.abi_method, context.method_args
+    )
+    note = None
+    if force_unique_transactions:
```

Comment:
If desired, get a different hash using a random note field. 80 bits of randomness should suffice to avoid flakiness.